### PR TITLE
Fix `feature = "nightly"` in the new trait solver

### PR DESCRIPTION
**Pull request by compiler-errors**
r? lcnr

### CODE REVIEW

File: library/std/Cargo.toml [hunk 1]
```diff
@@ -12,11 +10,11 @@ edition = "2021"
 crate-type = ["dylib", "rlib"]
 
 [dependencies]
-alloc = { path = "../alloc", public = true }
+alloc = { path = "../alloc" }
```
Comment (compiler-errors) [inline library/std/Cargo.toml:13]: fff I didn't mean to commit this!

File: compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs [hunk 2]
```diff
@@ -304,9 +304,10 @@ pub(in crate::solve) fn extract_tupled_inputs_and_output_from_callable<I: Intern
             let kind_ty = args.kind_ty();
             let sig = args.coroutine_closure_sig().skip_binder();
 
-            let coroutine_ty = if let Some(closure_kind) = kind_ty.to_opt_closure_kind()
-                && !args.tupled_upvars_ty().is_ty_var()
-            {
+            // FIXME: let_chains
+            let kind = kind_ty.to_opt_closure_kind();
+            let coroutine_ty = if kind.is_some() && !args.tupled_upvars_ty().is_ty_var() {
+                let closure_kind = kind.unwrap();
```
Comment (lcnr) [inline compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs:310]:
```suggestion
            let coroutine_ty = if let Some(kind) = kind_ty.to_opt_closure_kind().filter(|_| !args.tupled_upvars_ty().is_ty_var()) {
```

File: compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs [hunk 3]
```diff
@@ -411,10 +412,11 @@ pub(in crate::solve) fn extract_tupled_inputs_and_output_from_async_callable<I:
...
             let sig = args.coroutine_closure_sig().skip_binder();
             let mut nested = vec![];
-            let coroutine_ty = if let Some(closure_kind) = kind_ty.to_opt_closure_kind()
-                && !args.tupled_upvars_ty().is_ty_var()
-            {
-                if !closure_kind.extends(goal_kind) {
+
+            // FIXME: let_chains
+            let kind = kind_ty.to_opt_closure_kind();
+            let coroutine_ty = if kind.is_some() && !args.tupled_upvars_ty().is_ty_var() {
```
Comment (lcnr) [inline compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs:418]: same here

**Review by lcnr — COMMENTED** [review]
r=me after nits :(

File: compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs (hunk 2 above)
Comment (compiler-errors) [reply compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs:310]: I feel like this is so much harder to read

**Pull request merged**